src/generators/csharp/renderers/ClassRenderer.ts

```diff
@@ -0,0 +1,128 @@
+import { CSharpRenderer } from '../CSharpRenderer';
+import { ClassPreset, CommonModel, PropertyType } from '../../../models';
+import { DefaultPropertyNames, getUniquePropertyName } from '../../../helpers';
+import { pascalCase } from 'change-case';
+
+/**
+ * Renderer for CSharp's `struct` type
+ * 
+ * @extends CSharpRenderer
+ */
+export class ClassRenderer extends CSharpRenderer {
+  public async defaultSelf(): Promise<string> {
+    const content = [
+      await this.renderProperties(),
+      await this.runCtorPreset(),
+      await this.renderAccessors(),
+      await this.runAdditionalContentPreset(),
+    ];
+    
+    if (this.model.additionalProperties !== undefined || this.model.patternProperties !== undefined) {
+      this.addDependency('using System.Collections.Generic;');
+    }
+
+    const formattedName = this.nameType(this.model.$id);
+    return `public class ${formattedName} {
+${this.indent(this.renderBlock(content, 2))}
+}`;
+  }
+
+  async renderProperties(): Promise<string> {
+    const properties = this.model.properties || {};
+    const content: string[] = [];
+
+    for (const [propertyName, property] of Object.entries(properties)) {
+      const rendererProperty = await this.runPropertyPreset(propertyName, property);
+      content.push(rendererProperty);
+    }
+    
+    if (this.model.additionalProperties !== undefined) {
+      const propertyName = getUniquePropertyName(this.model, DefaultPropertyNames.additionalProperties);
+      const additionalProperty = await this.runPropertyPreset(propertyName, this.model.additionalProperties, PropertyType.additionalProperty);
+      content.push(additionalProperty);
+    }
+
+    if (this.model.patternProperties !== undefined) {
+      for (const [pattern, patternModel] of Object.entries(this.model.patternProperties)) {
+        const propertyName = getUniquePropertyName(this.model, `${pattern}${DefaultPropertyNames.patternProperties}`);
+        const renderedPatternProperty = await this.runPropertyPreset(propertyName, patternModel, PropertyType.patternProperties);
+        content.push(renderedPatternProperty);
+      }
+    }
+
+    return this.renderBlock(content);
+  }
+
+  async renderAccessors(): Promise<string> {
+    const properties = this.model.properties || {};
+    const content: string[] = [];
+    const accessorFactory = async (property: CommonModel, propertyName: string, type: PropertyType) => {
+      const formattedPropertyName = pascalCase(this.nameProperty(propertyName, property));
```

Comment:
Why do you wanna wrap the nameProperty with the `pascalCase`? If it's needed, It should be done inside `nameProperty` function.

Reply:
Also `accessorFactory` function could be written outside this function (as private function of Renderer), because as I see you don't use the closures.

Reply:
The reason for wrapping it, is because the accessor name (should always be upper-case first) and the actual property should be lowercase for the accessors to take effect. 
I.e. the following needs to be generated, notice camelcase property name and uppercase camelcase property name for the accessor.
```csharp
public class Address {
  private string streetName;
  public string StreetName 
  {
    get { return streetName; }
    set { streetName = value; }
  }
}
```

Reply:
@jonaslagoni It's a design pattern for C# or what?

Reply:
Not a design pattern per say, or is accessors a design pattern? 🧐 It is simply just the syntax

Reply:
I mean, should it be documented that we will generate accessors?

Reply:
We have this: https://github.com/jonaslagoni/generator-model-sdk/blob/feature/csharp/docs/presets.md#class-3

But it comes back to https://github.com/asyncapi/modelina/issues/318

Reply:
I mean, it is based on their official documentation - https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-properties#the-set-accessor 

It is just how they do it in C# 🤷